src/syscheckd/db/src/fimDB.cpp

```diff
@@ -166,3 +152,35 @@ void FIMDB::executeQuery(const nlohmann::json& item, ResultCallbackData callback
 {
     m_dbsyncHandler->selectRows(item, callbackData);
 }
+
+void FIMDB::fimRunIntegrity()
+{
+    std::unique_lock<std::mutex> lock{m_fimSyncMutex};
+
+    registerRSync();
+    loopRSync(lock);
```

Comment:
Where will this resource be unlocked? What happen if this generate a exception? I think that you should manage exceptions here!

Reply:
done, added error handling to functions in db.cpp

Reply:
This lock is unlocked in the wait_for function inside while loop, loopRSync function